File: php/reverse-string/ReverseString.php
<?php

declare(strict_types=1);

/**
 * Reverses a given string.
 *
 * This function takes a string, splits it into an array of individual characters,
 * reverses the order of those characters, and then joins them back into a single string.
 *
 * @param string $text The input string to be reversed.
 * @return string The reversed string.
 *
 * @example
 * reverseString("hello"); // Returns "olleh"
 * reverseString("world"); // Returns "dlrow"
 * reverseString("12345"); // Returns "54321"
 */
function reverseString(string $text): string
{
    return implode(array_reverse(str_split($text)));
}

File: csharp/reverse-string/ReverseString.cs
/// <summary>
/// A static class that provides a method for reversing a string.
/// </summary>
public static class ReverseString
{
    /// <summary>
    /// Reverses the order of the characters in a given string.
    /// </summary>
    /// <param name="input">The string to be reversed.</param>
    /// <returns>A new string with the characters from the input string in reverse order.</returns>
    public static string Reverse(string input)
       => new([.. input.Reverse()]);

}